tests/TestCase/src/Controller/ProductsFrontendControllerTest.php

```diff
@@ -125,4 +127,23 @@ protected function changeProductStatus($productId, $active)
         $statement = $this->dbConnection->prepare($query);
         $statement->execute($params);
     }
+
+    protected function changeProductDeliveryRhythm($productId, $deliveryRhythmType, $deliveryRhythmFirstDeliveryDay = '', $deliveryRhythmOrderPossibleUntil = '', $deliveryRhythmSendOrderListWeekday = '', $deliveryRhythmSendOrderListDay = '')
+    {
+        $this->configRequest([
+            'headers' => [
+                'Accept' => 'application/json',
+            ]
+        ]);
```

Comment:
please wrap that part in a new method `jsonPost`. ideally extend IntegrationTestCase or if that's too much for now just drop it in AppCakeTestCase.php